### PR TITLE
EVG-7334: clean up host provisioning

### PR DESCRIPTION
**Pull request by Kimchelly**
JIRA: https://jira.mongodb.org/browse/EVG-7334

This is an assortment of random, unrelated improvements and cleanup.

* Make convenience functions for running SSH commands. This includes running arbitrary scripts over SSH so there is no chance of shell quoting problems due to faulty argument interpretation.
* Make teardown more reliable by falling back to SSH in case Jasper fails.
* Print output of teardown script to stdout so it can be captured by host termination job.
* Consolidate common commands between setup host job and user data.
* Replace all usages of `*exec.Cmd` with `*jasper.Command`.
* Remove old init system detection code.
* Add TODO for fixing task fetching on Windows legacy SSH spawn hosts (none at this point, but might as well fix).

### CODE REVIEW

File: units/provisioning_setup_host.go [hunk 1]
```diff
@@ -589,309 +582,203 @@ func expandScript(s string, settings *evergreen.Settings) (string, error) {
...
-	}
-	if target.ProvisionOptions.OwnerId == "" {
-		return nil, errors.New("OwnerId not set")
-	}
-
-	// get the information about the owner of the host
-	owner, err := user.FindOne(user.ById(target.ProvisionOptions.OwnerId))
+// // loadClientResult indicates the locations on a target host where the CLI
+// // binary and its config file have been written to.
+// type loadClientResult struct {
```
Comment (john-m-liu) [inline units/provisioning_setup_host.go]: Delete this if it's not needed anymore

File: model/host/hostutil.go [hunk 2]
```diff
@@ -57,27 +55,27 @@ func (h *Host) TearDownCommand() string {
...
 
-func ShCommandWithSudo(ctx context.Context, script string, sudo bool) *exec.Cmd {
+func ShCommandWithSudo(script string, sudo bool) []string {
+	args := []string{}
 	if sudo {
-		return exec.CommandContext(ctx, "sudo", "sh", script)
+		args = append(args, "sudo")
 	}
-	return exec.CommandContext(ctx, "sh", script)
+	return append(args, "sh", script)
```
[the remark this answers is not in the document]
Comment (Kimchelly) [reply model/host/hostutil.go]: This is just the command. The context we passed into this has always been `context.Background()` from `operations/host_setup.go`, but this gets canceled anyways when SSH times out.

File: model/host/hostutil.go [hunk 3]
```diff
@@ -179,97 +177,67 @@ func (h *Host) GetSSHOptions(settings *evergreen.Settings) ([]string, error) {
...
-
-	return output.String(), errors.Wrap(err, "error running SSH command")
-}
-
-// InitSystem determines the current Linux init system used by this host.
-func (h *Host) InitSystem(ctx context.Context, sshOptions []string) (string, error) {
-	logs, err := h.RunSSHCommand(ctx, initSystemCommand(), sshOptions)
-	if err != nil {
-		return "", errors.Wrapf(err, "init system command returned: %s", logs)
+	if timeout != time.Duration(0) {
```
[the remark this answers is not in the document]
Comment (Kimchelly) [reply model/host/hostutil.go]: Removed type assertion.

File: model/host/hostutil.go [hunk 4]
```diff
@@ -1004,47 +1026,30 @@ func (h *Host) SetupSpawnHostCommands(settings *evergreen.Settings) (string, err
...
+// SpawnHostGetTaskDataCommand returns the command that fetches the task data
+// for a spawn host.
+func (h *Host) SpawnHostGetTaskDataCommand() []string {
+	// TODO (EVG-7387): this should be fixed to work with legacy SSH on Windows.
+	// The config file path has to be a native Windows path in order to properly
+	// read the config file, but currently the working directory is specified as
+	// a Cygwin-style directory (i.e. /home/Administrator should actually be
+	// C:/cygwin/home/Administrator).
+	return []string{h.AgentBinary(),
+		"-c", filepath.Join(h.Distro.BootstrapSettings.RootDir, h.spawnHostConfigFile()),
```
[the remark this answers is not in the document]
Comment (Kimchelly) [reply model/host/hostutil.go]: This won't work for user data, which runs as root on Linux and SYSTEM on Windows.

File: units/provisioning_setup_host.go (hunk 1 above)
Comment (Kimchelly) [reply units/provisioning_setup_host.go]: Done.

**Pull request merged**